src/function/scalar/list/list_contains.cpp

```diff
@@ -0,0 +1,226 @@
+#include <iostream>
+#include "duckdb/planner/expression/bound_function_expression.hpp"
+#include "duckdb/function/scalar/nested_functions.hpp"
+#include "duckdb/planner/expression_binder.hpp"
+
+namespace duckdb {
+
+static void SetResultFalse(Vector &result) {
+	auto result_data = FlatVector::GetData<bool>(result);
+	result_data[0] = false;
+	return;
+}
+
+template <class T>
+static void TemplatedListContainsStringFunction(DataChunk &args, ExpressionState &state, Vector &result) {
+	D_ASSERT(args.ColumnCount() == 2);
+	auto count = args.size();
+	Vector &list = args.data[0];
+	Vector &value = args.data[1];
+	VectorData value_data;
+	value.Orrify(count, value_data);
+
+	if (list.GetType().id() == LogicalTypeId::SQLNULL) {
+		SetResultFalse(result);
+		return;
+	}
+
+	auto list_size = ListVector::GetListSize(list);
+	if (list_size == 0) { // empty list will never contain a value
+		SetResultFalse(result);
+		return;
+	}
+	auto &child_vector = ListVector::GetEntry(list);
+	if (child_vector.GetType().id() == LogicalTypeId::SQLNULL) {
+		SetResultFalse(result);
+		return;
+	}
+	VectorData child_data;
+	child_vector.Orrify(list_size, child_data);
+
+	VectorData list_data;
+	list.Orrify(count, list_data);
+	auto list_entries = (list_entry_t *)list_data.data;
+
+	result.SetVectorType(VectorType::FLAT_VECTOR);
+	auto result_entries = FlatVector::GetData<bool>(result); // Create a vector of bool
+	auto &result_validity = FlatVector::Validity(result);
+
+	for (idx_t i = 0; i < count; i++) {
+		auto list_index = list_data.sel->get_index(i);
+
+		if (!list_data.validity.RowIsValid(list_index)) {
+			result_validity.SetInvalid(i);
+			continue;
+		}
+		result_entries[list_index] = false;
+
+		const auto &entry = list_entries[list_index];
+		auto source_idx = child_data.sel->get_index(entry.offset);
+		auto child_value = FlatVector::GetData<T>(child_vector);
+
+		for (idx_t child_idx = 0; child_idx < entry.length; child_idx++) {
+			auto value_idx = source_idx + child_idx;
+			if (!child_data.validity.RowIsValid(value_idx)) {
+				continue;
+			}
+			auto actual_value = child_value[value_idx];
+			if (StringComparisonOperators::EqualsOrNot<false>(actual_value, ((T *)value_data.data)[0])) {
```

Comment:
This line is the only difference between the templated and the string_t version of the code.

I think you can have only 1 `TemplatedListContainsFunction`, and use a templated comparison here. Then, all you need to do is write an explicit specialization for strings, e.g.:
```c++
template <class T>
static inline bool MyComparisonFunction(const T &left, const T&right) {
    return left == right;
}

template <>
inline bool MyComparisonFunction(const string_t &left, const string &right) {
    return StringComparisonOperators::EqualsOrNot<false>(left, right);
}
```

This saves a full duplication of the code.